protocol/protocol.tex

```diff
@@ -9467,37 +11392,98 @@
 
-$192$ & $\zkproof$ & \type{char[192]} & An encoding of the \zkSNARKProof
+$192$ & $\zkproof$ & \type{byte[192]} & An encoding of the \zkSNARKProof
 $\ProofOutput$ (see \crossref{groth}). \\ \hline
 
 \end{tabularx}
 \end{center}
 
 \vspace{-2ex}
 The $\ephemeralKey$, $\encCiphertext$, and $\outCiphertext$ fields together form the
-\noteCiphertext, which is computed as described in \crossref{saplinginband}.
+\noteCiphertextSapling, which is computed as described in \crossref{saplinginband}.
 
 \vspace{-2ex}
 \consensusrule{$\LEOStoIPOf{256}{\cmuField}$ \MUST be less than $\ParamJ{q}$.}
 
 \vspace{-0.5ex}
 Other consensus rules applying to an \outputDescription are given in \crossref{outputdesc}.
-}
+} %sapling
+
+
+\orchard{
+\introsection
+\lsubsection{Action Description Encoding and Consensus}{actionencodingandconsensus}
+
+Let $\LEBStoOSP{}{}$ be as defined in \crossref{endian}.
+
+\vspace{-0.5ex}
+Let $\reprP$ and $\ParamP{q}$ be as defined in \crossref{pallasandvesta}.
+
+\vspace{-0.5ex}
+An abstract \actionDescription, as described in \crossref{actions}, is encoded in
+a \transaction as an instance of an \type{ActionDescription} type as follows:
+
+\vspace{-2.5ex}
+\begin{center}
+\hbadness=2000
+\begin{tabularx}{0.92\textwidth}{|c|l|l|L|}
+\hline
+Bytes & \heading{Name} & \heading{Data Type} & \heading{Description} \\
+\hhline{|=|=|=|=|}
+
+$32$ & $\cvField$ & \type{byte[32]} & A \valueCommitment to the net value of the input \note
+minus the output \note, $\LEBStoOSPOf{256}{\reprP\Of{\cv}}$. \\ \hline
+
+$32$ & $\nullifierField$ & \type{byte[32]} & The \nullifier of the input \note, $\nf$. \\ \hline
+
+$32$ & $\rkField$ & \type{byte[32]} & The randomized \validatingKey for $\spendAuthSig$,
+$\LEBStoOSPOf{256}{\reprJ\Of{\AuthSignRandomizedPublic}\kern 0.05em}$. \\ \hline
+
+$32$ & $\cmuField$ & \type{byte[32]} & The $u$-coordinate of the \noteCommitment for the output \note,
+$\LEBStoOSPOf{256}{\cmU}$ where $\cmU = \ExtractJ(\cm)$. \\ \hline
+
+$32$ & $\ephemeralKey$ & \type{byte[32]} & An encoding of an ephemeral \Jubjub \publicKey,
+$\LEBStoOSPOf{256}{\reprJ\Of{\EphemeralPublic}}$. \\ \hline
+
+$580$ & $\encCiphertext$ & \type{byte[580]} & A ciphertext component for the
+encrypted output \note, $\TransmitCiphertext{}$. \\ \hline
+
+$80$ & $\outCiphertext$ & \type{byte[80]} & A ciphertext component for the
+encrypted output \note, $\OutCiphertext{}$. \\ \hline
+
+$2208$ & $\zkproof$ & \type{byte[2208]} & An encoding of the \zkSNARKProof
+$\ProofAction$ (see \crossref{halo2}). \\ \hline
+
+$64$ & $\spendAuthSig$ & \type{byte[64]} & A signature authorizing this Spend. \\ \hline
+
+\end{tabularx}
+\end{center}
+
+\vspace{-2ex}
+The $\ephemeralKey$, $\encCiphertext$, and $\outCiphertext$ fields together form the
+\noteCiphertextOrchard, which is computed as described in \crossref{saplinginband}.
+
+\vspace{-2ex}
+\consensusrule{$\LEOStoIPOf{256}{\cmxField}$ \MUST be less than $\ParamP{q}$.}
```

Comment:
There doesn't seem to be a `cmx` field here:
```suggestion
\consensusrule{$\LEOStoIPOf{256}{\cmuField}$ \MUST be less than $\ParamP{q}$.}
```

Reply:
`cmx` is correct. (The error was elsewhere.)